hil/api.py

```diff
@@ -1099,13 +1115,40 @@ def port_revert(switch, port):
     if port.nic is None:
         raise errors.NotFoundError(port.label + " not attached")
     if port.nic.current_action:
-        raise errors.BlockedError("Port already has a pending action.")
+        if port.nic.current_action.status == "PENDING":
+            raise errors.BlockedError("Port already has a pending action.")
+        else:
+            db.session.delete(port.nic.current_action)
 
-    db.session.add(model.NetworkingAction(type='revert_port',
-                                          nic=port.nic,
-                                          channel='',
-                                          new_network=None))
+    unique_id = str(uuid.uuid4())
+
+    action = model.NetworkingAction(type='revert_port',
+                                    nic=port.nic,
+                                    channel='',
+                                    uuid=unique_id,
+                                    status='PENDING',
+                                    new_network=None)
+
+    db.session.add(action)
     db.session.commit()
+    return json.dumps({'status_id': unique_id})
+
+
+# is this only going to be for statuses of Networking actions? What about in
+# the future we have some other calls whose status we want to check?
+@rest_call('GET', '/status/<status_id>', Schema({'status_id': basestring}))
+def get_status(status_id):
+    """Returns the status of the networking action by findind the status_id
+    in the networking actions table
+    """
+    # is auth required here? UUID is obscure enough that nobody can guess
+    # and see someone else's status of their operations. Moreover, it doesn't
```

Comment:
The uuid generation we're using here is *not* cryptographically random, so we should not assume it is unguessable.

I'd rather play it safe and do an authorization check than fuss about whether the information is sensitive. Probably the appropriate thing to check is either admin access or access to the project of the attached node.

Reply:
> Probably the appropriate thing to check is either admin access or access to the project of the attached node.

Okay, ill do a check for access to project.